src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,42 @@
-'use strict';
-
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  die() {
+    const index = Animal.alive.indexOf(this);
+
+    if (index !== -1) {
+      Animal.alive.splice(index, 1);
+    }
```

Comment:
[CODE PERFORMANCE] - The current implementation of the die method uses indexOf and splice within a loop, which can be inefficient. Consider using filter to remove dead animals from the Animal.alive array. For example, you could replace the die method with a static method that filters the Animal.alive array for animals with health greater than 0.